src/snapshot_derive/src/common.rs

```diff
@@ -0,0 +1,53 @@
+use quote::format_ident;
+use std::collections::hash_map::HashMap;
+
+// Returns a string literal attribute as an Ident.
+pub(crate) fn get_ident_attr(
+    attrs: &HashMap<String, syn::Lit>,
+    attr_name: &str,
+) -> Option<syn::Ident> {
+    attrs.get(attr_name).map(|default_fn| match default_fn {
+        syn::Lit::Str(lit_str) => {
+            return format_ident!("{}", lit_str.value());
+        }
+        _ => panic!("default_fn must be the function name as a String."),
+    })
+}
+
+// Parses field annotations.
+pub(crate) fn parse_field_attributes(
+    attrs: &mut HashMap<String, syn::Lit>,
+    attributes: &Vec<syn::Attribute>,
+) {
+    for attribute in attributes {
+        // Check if this is a snapshot attribute.
+        match attribute.parse_meta().unwrap().clone() {
+            syn::Meta::List(meta_list) => {
+                // Check if this is a "snapshot" attribute.
+                if meta_list.path.segments[0].ident.to_string() == "snapshot" {
+                    // Fetch the specific attribute name
+                    for nested_attribute in meta_list.nested {
+                        match nested_attribute {
+                            syn::NestedMeta::Meta(nested_meta) => {
+                                match nested_meta {
+                                    syn::Meta::NameValue(attr_name_value) => {
+                                        // panic!("{:?}", attr_name_value);
+                                        // if attr_name_value.eq_token.to_string() == "=" {
+                                        attrs.insert(
+                                            attr_name_value.path.segments[0].ident.to_string(),
+                                            attr_name_value.lit,
+                                        );
+                                        // }
+                                    }
+                                    _ => {}
+                                }
+                            }
+                            _ => {}
+                        }
+                    }
+                }
+            }
+            _ => {}
+        }
+    }
+}
```

Comment:
I find it easier to follow nested `if` statements than nested `match` statements here. Since there would be no `else` branch.

Reply:
I will give this a try to see how it looks.

Reply:
Actually how about something like:

```
    for nested_meta in attributes
        .iter()
        .flat_map(|attr| -> Result<Vec<syn::NestedMeta>, ()> {
            if attr.path.is_ident("snapshot") {
                return Ok(Vec::new());
            }

            if let Ok(syn::Meta::List(meta)) = attr.parse_meta() {
                return Ok(meta.nested.into_iter().collect());
            }

            return Ok(Vec::new());
        })
        .flatten()
    {
        if let syn::Meta::NameValue(attr_name_value) = nested_meta {
            attrs.insert(
                attr_name_value.path.segments[0].ident.to_string(),
                attr_name_value.lit,
            );
        }
    }
```